libctru/include/3ds/svc.h

```diff
@@ -930,11 +988,19 @@ Result svcKernelSetState(u32 type, ...);
 void svcBreak(UserBreakType breakReason);
 
 /**
+ * @brief Breaks execution.
+ * @param breakReason Debug reason for breaking.
+ * @param debugParameter1 First debugging parameter
+ * @param debugParameter2 Second debugging parameter
+ */
+void svcDebugBreak(UserBreakType breakReason, u32 debugParameter1, u32 debugParameter2);
+
+/**
  * @brief Outputs a debug string.
  * @param str String to output.
- * @param length Length of the string to output.
+ * @param length Length of the string to output (needs to be positive).
  */
-Result svcOutputDebugString(const char* str, int length);
+Result svcOutputDebugString(const char* str, s32 length);
```

Comment:
Signed integers don't have a "sign bit" per se. It just happens to be that the high bit of a negative number is always set, but that's just a side-effect of twos-complement form. If they truly had a sign bit, you would gain -0 but lose -2147483648.

In any case, the kernel treats this parameter as signed. I would therefore leave the size parameter as s32.